test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -1090,8 +1192,102 @@ private void checkWindowsLists() {
             }
         }
 
-        public Builder positionTestUI(PositionWindows positionWindows) {
-            this.positionWindows = positionWindows;
+        /**
+         * Adds a {@code PanelCreator} which the framework will use
+         * to create a component and place it into a dialog.
+         *
+         * @param panelCreator a {@code PanelCreator} to create a component
+         *                     with test UI
+         * @return this builder
+         * @throws IllegalStateException if split UI was enabled using
+         *              a {@code splitUI} method
+         */
+        public Builder testUI(PanelCreator panelCreator) {
+            if (splitUI) {
+                throw new IllegalStateException("Can't combine splitUI and "
+                                                + "testUI with panelCreator");
+            }
+            this.panelCreator = panelCreator;
+            return this;
+        }
+
+        /**
+         * Adds a {@code PanelCreator} which the framework will use
+         * to create a component with test UI and display it in a split pane.
+         * <p>
+         * By default, horizontal orientation is used,
+         * and test UI is displayed to the right of the instruction UI.
+         *
+         * @param panelCreator a {@code PanelCreator} to create a component
+         *                     with test UI
+         * @return this builder
+         *
+         * @throws IllegalStateException if a {@code PanelCreator} is
+         *              already set
+         * @throws IllegalArgumentException if {panelCreator} is {@code null}
+         */
+        public Builder splitUI(PanelCreator panelCreator) {
```

Comment:
```suggestion
        public Builder splitUI(PanelCreator panelCreator, int splitUIOrientation ) {
```


Would it be easier if we use `splitUIOrientation` or an enum similar to [Position](https://github.com/openjdk/jdk/blob/b823fa44508901a6bf39795ab18991d055a71b4e/test/jdk/java/awt/regtesthelpers/PassFailJFrame.java#L191) to determine the split than have three different methods - `splitUI, splitUIRight & splitUIBottom`?

Reply:
In terms of flexibility & extensibility your approach works better.